src/ucp/rndv/rndv_get.c

```diff
@@ -171,47 +171,6 @@ ucp_proto_rndv_get_zcopy_fetch_progress(uct_pending_req_t *uct_req)
             ucp_proto_rndv_get_zcopy_fetch_completion);
 }
 
-static void
-ucp_proto_rndv_get_zcopy_fetch_err_completion(uct_completion_t *uct_comp)
-{
-    ucp_request_t *req  = ucs_container_of(uct_comp, ucp_request_t,
-                                          send.state.uct_comp);
-
-    ucp_datatype_iter_mem_dereg(&req->send.state.dt_iter,
-                                UCS_BIT(UCP_DATATYPE_CONTIG));
-    ucp_proto_rndv_rkey_destroy(req);
-    ucp_proto_rndv_recv_complete_status(req, uct_comp->status);
-}
-
-static void ucp_rndv_get_zcopy_proto_abort(ucp_request_t *request,
-                                           ucs_status_t status)
-{
-    ucp_request_t *rreq UCS_V_UNUSED;
-
-    switch (request->send.proto_stage) {
-    case UCP_PROTO_RNDV_GET_STAGE_FETCH:
-        /* The error completion handler is not sending ATS */
-        request->send.state.uct_comp.func =
-                ucp_proto_rndv_get_zcopy_fetch_err_completion;
-        ucp_proto_request_zcopy_abort(request, status);
-        break;
-    case UCP_PROTO_RNDV_GET_STAGE_ATS:
```

Comment:
If memory invalidation works properly with this patch, can we return success if `abort()` was called on the ATS stage? I mean all data is received and since there is abort with shouldn't send ATP but at least can set status of this request to UCS_OK, is that right? @evgeny-leksikov @yosefe

Reply:
IMHO, this is not mandatory, it's just best effort to complete request successfully, in case of connection failure we can complete the request with error. Moreover, so far request restart is not implemented for intermediate stage, it might be simpler to fail the request since sender may try to re-transmit it.

Reply:
IMO, either way is fine (whichever leads to simpler code). And put a comment that documents the behavior.

Reply:
I decided to return UCS_OK in that case. Tested through io_demo. Seems ok.